quickwit/quickwit-cluster/src/cluster.rs

```diff
@@ -454,13 +454,51 @@ fn parse_enabled_services_str(
     Ok(enabled_services)
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+// Not used within the code, used for documentation.
+#[derive(Debug, utoipa::ToSchema)]
+pub struct NodeIdSchema {
+    #[schema(example = "node-1")]
+    /// The unique identifier of the node in the cluster.
+    pub id: String,
+
+    #[schema(example = "127.0.0.1:8000", value_type = String)]
+    /// The SocketAddr other peers should use to communicate.
+    pub gossip_public_address: SocketAddr,
+}
+
+#[derive(Serialize, Deserialize, Debug, utoipa::ToSchema)]
 pub struct ClusterSnapshot {
+    #[schema(example = "qw-cluster-1")]
+    /// The ID of the cluster tha the node is a part of.
     pub cluster_id: String,
+
+    #[schema(value_type = NodeIdSchema)]
+    /// The unique ID of the current node.
     pub self_node_id: NodeId,
+
+    #[schema(
+        value_type = Object,
+        example = json!({
+            "key_values": {
+                "grpc_advertise_addr": "127.0.0.1:8080",
+                "enabled_services": "",
+            },
+            "max_version": 5,
+        })
+    )]
+    /// A snapshot of the current cluster state.
     pub chitchat_state_snapshot: ClusterStateSnapshot,
+
+    #[schema(value_type  = Vec<NodeIdSchema>)]
+    /// The set of node IDs that are ready to handle operations.
     pub ready_nodes: HashSet<NodeId>,
+
+    #[schema(value_type  = Vec<NodeIdSchema>)]
+    /// The set of node IDs that are online and a part of the cluster.
```

Comment:
I would stick to "live" word.

```suggestion
    /// The set of cluster node IDs that are alive.
```